kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -507,6 +507,10 @@ protected void validateWriteToExistingTable(Engine engine, SnapshotImpl snapshot
                 txnId.getAppId(), txnId.getVersion(), lastTxnVersion.get());
           }
         });
+    if (clusteringColumns.isPresent()
+        && snapshot.getMetadata().getPartitionColumns().getSize() != 0) {
+      throw new KernelException("Cannot set clustering columns on a partitioned table");
```

Comment:
add this to DeltaErrors? And maybe include more information (existing partition columns, clustering columns set?)

Reply:
Also maybe be more specific? if someone does `clusterBy(emptyList())` this should fail right? based on the docs for `withClusteringColumns` this makes it a clustered table

So maybe "cannot enable clustering on a partitioned table" is more accurate?